File: scholarzim/src/main/java/com/scholarzim/entity/ProviderProfile.java
package com.scholarzim.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;


@Entity
@Table(name = "provider_profiles")
@Getter
@Setter
public class ProviderProfile {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "profile_id")
    private Long profileId;

    @OneToOne
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;

    @Column(name = "organisation_type", nullable = false)
    private String organisationType;

    @Column(name = "registration_number", nullable = false)
    private String registrationNumber;

    @Column(name = "certificate_path", nullable = false)
    private String certificatePath;

    @Column(name = "certificate_filename", nullable = false)
    private String certificateFilename;

    @Column(name = "submitted_at", nullable = false)
    private LocalDateTime submittedAt;

    @Column(name = "reviewed_at")
    private LocalDateTime reviewedAt;

    @Column(name = "reviewed_by")
    private String reviewedBy;

    @Column(name = "rejection_reason", length = 500)
    private String rejectionReason;
}

File: app/Models/ProviderProfile.php
<?php

namespace App\Models;

use App\Support\ProviderOrgType;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class ProviderProfile extends Model
{
    protected $table = 'provider_profiles';

    protected $primaryKey = 'profile_id';

    public $timestamps = false;

    protected $fillable = [
        'user_id',
        'organisation_type',
        'registration_number',
        'certificate_path',
        'certificate_filename',
        'submitted_at',
        'reviewed_at',
        'reviewed_by',
        'rejection_reason',
    ];

    protected $casts = [
        'submitted_at' => 'datetime',
        'reviewed_at' => 'datetime',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class, 'user_id', 'user_id');
    }

    public function organisationTypeLabel(): string
    {
        return ProviderOrgType::label($this->organisation_type);
    }

    public function isReviewed(): bool
    {
        return $this->reviewed_at !== null;
    }
}
